src/RequestBuilder/RecommendationRequestBuilder.php

```diff
@@ -58,13 +58,24 @@ public function build(): Request
 
     private function assertConsistentUsersInCommands(): void
     {
-        $mainCommandUser = $this->userRecommendationCommand->getUserId();
-        if ($this->interactionCommand !== null && $mainCommandUser !== $this->interactionCommand->getUserId()) {
-            throw LogicException::forInconsistentUserId($this->userRecommendationCommand, $this->interactionCommand);
+        $interactionUser = $this->interactionCommand ? $this->interactionCommand->getUserId() : null;
+        $mergeFromUser = $this->userMergeCommand ? $this->userMergeCommand->getSourceUserId() : null;
+        $mergeIntoUser = $this->userMergeCommand ? $this->userMergeCommand->getUserId() : null;
+        $recommendationUser = $this->userRecommendationCommand->getUserId();
+
+        // ([A], [A -> ?], ?)
+        if ($this->interactionCommand !== null && $this->userMergeCommand !== null && $interactionUser !== $mergeFromUser) {
```

Comment:
I would separate those three conditions inside `if` to a private method (for better readability).

Similary in the cases below (and in other Builders).